src/python-frontend/python_converter.cpp

```diff
@@ -433,21 +461,51 @@ exprt python_converter::get_expr(const nlohmann::json &element)
   }
   case ExpressionType::VARIABLE_REF:
   {
-    // Find the variable declaration in the current function
-    std::string var_name = element["id"].get<std::string>();
-    std::string symbol_id = "py:" + python_filename + "@" + "F" + "@" +
-                            current_func_name + "@" + var_name;
+    std::string var_name;
+    if (element["_type"] == "Name")
+      var_name = element["id"].get<std::string>();
+    else if (element["_type"] == "Attribute")
+      var_name = element["value"]["id"].get<std::string>();
+
+    assert(!var_name.empty());
+
+    std::string symbol_id = create_symbol_id() + std::string("@") + var_name;
     symbolt *symbol = context.find_symbol(symbol_id);
-    if (symbol != nullptr)
-    {
-      expr = symbol_expr(*symbol);
-    }
-    else
+    if (!symbol)
     {
-      log_error("Symbol not found\n");
+      log_error("Symbol not found: {}\n", symbol_id.c_str());
       abort();
     }
+    expr = symbol_expr(*symbol);
+
+    if (element["_type"] == "Attribute")
+    {
+      const std::string &attr_name = element["attr"].get<std::string>();
+
+      // Get object type name from symbol. e.g.: tag-MyClass
+      std::string obj_type_name;
+      const typet &symbol_type =
+        (symbol->type.is_pointer()) ? symbol->type.subtype() : symbol->type;
+      for (const auto &it : symbol_type.get_named_sub())
+      {
+        if (it.first == "identifier")
+          obj_type_name = it.second.id_string();
+      }
+      assert(!obj_type_name.empty());
```

Comment:
Shall we really use assert statements here? What happens with the release mode when we disable these assertions?

Reply:
In release we would have a crash when dereferecing class_symbol at https://github.com/brcfarias/esbmc/blob/69112185c12d40be1eed5d87ea57c4eb6e9e23ac/src/python-frontend/python_converter.cpp#L502C1-L503C1.

I will remove the asserts, check if class_symbol is valid before dereferencing, and call `abort()` if so.

Reply:
Excellent! Thanks, @brcfarias.